File: examples/python/core/callbacks/conditional-callback-handling.py
@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Handle callback results conditionally."""
    callback = context.create_callback(
        name="conditional_callback",
        config=CallbackConfig(timeout=Duration.from_minutes(10)),
    )
    
    # Send request
    send_request(callback.callback_id, event["request_type"])
    
    # Wait for result
    result = callback.result()
    
    # Handle different result types
    if result is None:
        return {"status": "timeout", "message": "No response received"}
    
    result_type = result.get("type")
    
    if result_type == "success":
        return process_success(result)
    elif result_type == "partial":
        return process_partial(result)
    else:
        return process_failure(result)

File: examples/java/core/callbacks/conditional-callback-handling.java
var callback = ctx.createCallback("conditional_callback", Map.class,
    CallbackConfig.builder()
        .timeout(Duration.ofMinutes(10))
        .build());

// Send request
ctx.step("send_request", Void.class, stepCtx -> {
    sendRequest(callback.callbackId(), event.get("request_type"));
    return null;
});

// Wait for result
Map<String, Object> result;
try {
    result = callback.get();
} catch (CallbackTimeoutException e) {
    return Map.of("status", "timeout", "message", "No response received");
}

// Handle different result types
var resultType = (String) result.get("type");

if ("success".equals(resultType)) {
    return processSuccess(result);
} else if ("partial".equals(resultType)) {
    return processPartial(result);
} else {
    return processFailure(result);
}
